pymeasure/instruments/siglenttechnologies/siglent_spd1305x.py

```diff
@@ -36,10 +36,10 @@ def __init__(self, adapter, **kwargs):
             name="Siglent Technologies SPD1305X Power Supply",
             **kwargs
         )
-        voltage_limits = [0, 30]
-        current_limits = [0, 5]
+        voltages = [0, 30]
+        currents = [0, 5]
 
-        self.ch[1] = SPDChannel(self, 1, voltage_limits, current_limits)
+        self.add_child(SPDChannel, 1, voltage_range=voltages, current_range=currents)
```

Comment:
We made a `ChannelCreator`, which works similar to the property creators, such that you do not need to add it to the init.
Instead you use it at class level:
~~~python
class xy(Instrument):
    channels = ChannelCreator(SPDChannel, 1, **kwargs)
~~~

The result will be, as if you called add_child at the beginning. `kwargs` is the dictionary of kwargs for the Channel initialization.

Reply:
I've got a couple of questions about how to move forward with this approach. Ive taken on board the suggestions from your last few comments here and have created the channels in SPDSingleChannelBase as so:
```python
class SPDSingleChannelBase(SPDBase):
    channels = Instrument.ChannelCreator(SPDChannel, 1)
```
but am running into a few issues. Firstly, you've included `**kwargs` in your example above. Why? This will be undefined at the class level. It would be possible if the channels object is overloaded in the `__init__` of the child instrument but this is only useful in the example for the 1305 instrument where the voltage and current range parameters are needed.

Secondly, this implementation consistently causes the tests to fail with the following error:
```python
FAILED tests/instruments/siglenttechnologies/test_siglent_spd1168x.py::test_set_current
AttributeError: 'SPD1168X' object has no attribute 'ch_1'
```
obviously this means that the channel init is failing somewhere but I can't seem to find the answer. Any ideas?

Reply:
Sorry for that `**kwargs`. I meant it as a placeholder for the arguments you want to give to the Channel (e.g. The limits), but not literally.

I have to look at the code in detail.

Reply:
Let me commit what I've got so far. It's not working, but at least you might have a chance to see what's going on

Reply:
Thanks for pushing.

I found the error: Right now, the automatic recognition only works, if the ChannelCreator is used in the last class, i.e. here in the SPD1168 class, not in the parent class, as we look just in the last class for ChannelCreators.

Therefore, you have to define `channels = ChannelCreator(...` in the SPD1168, not in the SingleClass (for now), then the tests succeed.
I'll open an issue and look into it.

Thank you for helping finding this bug.